### PR TITLE
320 rounds: feedback for two stage catchpoints

### PR DESCRIPTION
**Pull request by tolikzinovyev**
## Summary

Address Chris's feedback for two stage catchpoints PR #3928.

## Test Plan

None.

### CODE REVIEW

File: ledger/catchpointtracker.go [hunk 1]
```diff
@@ -474,14 +476,8 @@ func doRepackCatchpoint(header CatchpointFileHeader, in *tar.Reader, out *tar.Wr
...
 
-		buf := make([]byte, header.Size)
-		bytesRead := int64(0)
-		for (err == nil) && (bytesRead < header.Size) {
-			var x int
-			x, err = in.Read(buf[bytesRead:])
-			bytesRead += int64(x)
-		}
-		if (err != nil) && (err != io.EOF) {
+		buf, err := io.ReadAll(in)
```
Comment (cce) [inline ledger/catchpointtracker.go:479]: I was thinking about this loop and how much GC pressure it makes, how many files/chunks are in a typical catchpoint file? We could move creating `buf` to be outside of the loop and re-used for each chunk file. We could remember what the max capacity would be needed for this bufer needs (from stage 1) and then reset the slice for each iteration `iterBuf = buf[:desiredLen]` to reset length but keep the capacity?

File: catchup/ledgerFetcher.go [hunk 2]
```diff
@@ -165,21 +165,9 @@ func (lf *ledgerFetcher) getPeerLedger(ctx context.Context, peer network.HTTPPee
...
-			if err != nil {
-				if err == io.EOF {
-					if readComplete == header.Size {
-						break
-					}
-					err = fmt.Errorf("getPeerLedger received io.EOF while reading from tar file stream prior of reaching chunk size %d / %d", readComplete, header.Size)
-				}
-				return err
-			}
+		balancesBlockBytes, err := io.ReadAll(tarReader)
```
Comment (cce) [inline catchup/ledgerFetcher.go:168]: if you are not going to `make([]byte, header.Size)` with io.ReadFull and use ReadAll instead, maybe add a manual check that len(balancesBlockBytes) == header.Size? and/or a check that len(balancesBlockSize) or header.Size != 0?

Comment (tolikzinovyev) [reply catchup/ledgerFetcher.go]: You are right. We probably shouldn't trust that the tar reader will not read more bytes than in the header if the file is malformed.

File: ledger/catchpointtracker.go (hunk 1 above)
Comment (tolikzinovyev) [reply ledger/catchpointtracker.go:479]: It's an interesting idea. Do you think this might be the cause for high memory usage in Pavel's tests?

Comment (cce) [reply ledger/catchpointtracker.go:479]: It is a lot of stuff to allocate and destroy all at once.. POC re-use commit https://github.com/cce/go-algorand/commit/9fdebcfd8df94f0aac1885ef00ca6d5dc613def2

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**